compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/model/symbols/BPackageSymbol.java

```diff
@@ -47,6 +48,7 @@
     public CompiledPackage compiledPackage;
     public Name compUnit;
     public boolean isUsed = false;
+    private ModuleContext moduleContext;
```

Comment:
We can make this public and remove the getter and setter

Reply:
I'm not sure why should we do that?

Reply:
@pubudu91

Reply:
Generally we have kept the fields public. Not exactly sure why it was done that way. But it's ok to keep these private. 

On a separate note, I don't feel like adding the module context to the module symbol is the correct approach to take. My suggestion is,
- Set the module context to the AST when creating the AST (`BLangPackage`)
- There's a place in `SymbolEnter` where we create a symbol (`BPackageSymbol`) for the module. At that point, set any info you need to access from the symbol to the symbol. e.g., whether this module is exported or not, the name of the package this module belongs to
- Persist these additional info you added to the symbol in the BIR
- Need to add a corresponding section in `BIRPackageSymbolEnter` where you read these additional info and set them to the `BPackageSymbol` you are creating for the module loaded from a BIR